src/compiler/types.ts

```diff
@@ -1493,6 +1499,8 @@ module ts {
     // Class and interface types (TypeFlags.Class and TypeFlags.Interface)
     export interface InterfaceType extends ObjectType {
         typeParameters: TypeParameter[];           // Type parameters (undefined if non-generic)
+        outerTypeParameters: TypeParameter[];      // Outer type parameters (undefined if none)
+        localTypeParameters: TypeParameter[];      // Local type parameters (undefined if none)
     }
 
     export interface InterfaceTypeWithBaseTypes extends InterfaceType {

```

Comment:
We always parse all modifiers on a declaration and then issue errors later during type check. We need to do this so our incremental parsing always parses declarations the same. Also, it provides better error handling.